repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/DefaultRepository.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (C) 2018-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.repository.kafka;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import lombok.RequiredArgsConstructor;
+
+import com.expediagroup.streamplatform.streamregistry.state.EntityView;
+import com.expediagroup.streamplatform.streamregistry.state.EventSender;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity;
+import com.expediagroup.streamplatform.streamregistry.state.model.event.Event;
+import com.expediagroup.streamplatform.streamregistry.state.model.status.StatusEntry;
+
+@RequiredArgsConstructor
+abstract class DefaultRepository<
+    ME extends com.expediagroup.streamplatform.streamregistry.model.Entity<MK>,
+    MK,
```

Comment:
Prefer more descriptive names here

Reply:
I've added javadoc on DefaultRepository to explain generics - good enough?

```
/**
 * @param <ME> Model Entity
 * @param <MK> Model Key
 * @param <SK> State Key
 * @param <SS> State Specification
 */
```

Reply:
Would prefer the generic names expanded, but happy with this.